packages/plugin-core/src/commands/VaultAddCommand.ts

```diff
@@ -133,6 +133,83 @@ export class VaultAddCommand extends BasicCommand<CommandOpts, CommandOutput> {
     };
   }
 
+  async gatherVaultSelfContained(
+    sourceType: VaultRemoteSource
+  ): Promise<CommandOpts | undefined> {
+    // If the vault name already exists, creating a vault with the same name would break things
+
+    if (sourceType === "local") {
```

Comment:
Instead of using naked strings, can we use enums? This makes refactoring easier when its used in many places